src/main/java/org/primefaces/extensions/renderer/CommandButtonSingleClickRenderer.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2011-2019 PrimeFaces Extensions
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.primefaces.extensions.renderer;
+
+import java.io.IOException;
+import javax.faces.context.FacesContext;
+import org.primefaces.component.commandbutton.CommandButton;
+import org.primefaces.component.commandbutton.CommandButtonRenderer;
+
+/**
+ * {@link CommandButton} renderer disabling the button while action is processed for buttons that are using Ajax.
+ *
+ * @author Jasper de Vries &lt;jepsar@gmail.com&gt;
+ * @since 8.0
+ */
+public class CommandButtonSingleClickRenderer extends CommandButtonRenderer {
+
+    @Override
+    protected void encodeMarkup(FacesContext context, CommandButton button) throws IOException {
+        if (isToggleable(button)) {
+            String widgetVar = button.resolveWidgetVar(context);
+            button.setOnclick(prefix(button.getOnclick(), toggle(widgetVar, false)));
+            button.setOncomplete(prefix(button.getOncomplete(), toggle(widgetVar, true)));
+        }
+        super.encodeMarkup(context, button);
+    }
+
+    protected boolean isToggleable(final CommandButton button) {
```

Comment:
I don't understand what you mean by `isToggleable`? Also, you can simplify your code with:
```java
return button.isRendered() && !button.isDisabled() && !isConfirmation(button) && button.isAjax()
```

Reply:
`isToggleable` will tell you if we can toggle enabled / disabled on it. I moved it to a method to prevent an if with the mother of all conditions. I think this is more readable.

Reply:
@rapster if you have a better name we are all ears!

Reply:
I don't know @melloware, I only asked to understand but maybe something like `isEligible`/`isEligibleToSingleClick`

Reply:
`isEligible` it is!